src/deserialize_sequence.h

```diff
@@ -37,6 +37,17 @@
 
 namespace task {
 
+/**
+ * Deserialize parameters of Sequence from the input stream.
+ *
+ * No checking of any stream failure is done and should be performed by the caller.
+ *
+ * \param stream input stream
+ * \param sequence Sequence to be deserialized.
+ * \return passed input stream
+ */
+std::istream& operator>>(std::istream& stream, Sequence& sequence);
```

Comment:
To me this gives the false impression that the whole sequence can be loaded from an input stream, but it cannot. This operator can only read a very tiny part of the `Sequence` object from the `sequence.lua` file, and it does not import any steps at all.

IMO we should state this **very** clearly in the function description or avoid an operator overload for this. My preference would be the latter.

This operator would work great if we had a separate class like `SequenceParameters` (excluding the steps), but we don't.

Reply:
This code snippet is only implemented for testing purposes but I agree to your comment. Sure, I can push it back to the `load_sequences_parameter(...)` but then it is not testable. Maybe we should extract the different properties of the Sequence to separate functions, for example for `autorun`: `parse_autorun(std::istream&): bool`...?

To have an extra class for the sequence parameter is a better solution but I think it belongs to another PR because I wanted to focus on the additional autorun parameter.

Reply:
> I can push it back to the load_sequences_parameter(...) but then it is not testable

It does not have to be inside `void load_sequence_parameters(const std::filesystem::path& folder, Sequence& sequence)`, but nobody is stopping us from making an overload `void load_sequence_parameters(std::istream& stream, Sequence& sequence)` and testing that.

Reply:
I move the serializartion of the sequence parameters to `void load_sequence_parameters(const std::filesystem::path& folder, Sequence& sequence)` and avoid the `operator>>(..., Sequence&)`.